### ISSUE TITLE
Problems with latest update

### ISSUE THREAD

**Issue by randymassey** [hbblebc/homebridge-bed-control#9]
**Describe The Bug:**
After updating the plugin generates errors.

**To Reproduce:**
Update to latest version...  If I backstep to prior version, it boots as expected.

**Expected behavior:**
Should boot child bridge up normally.
**Logs:**

```

[09/06/2024, 12:07:46] [homebridge-bed-control] Restarting Process...
[09/06/2024, 12:07:47] [homebridge-bed-control] Launched child bridge with PID 6795
Error: Cannot find module 'hap-nodejs/dist/lib/definitions'
Require stack:
- /var/lib/homebridge/node_modules/homebridge-bed-control/dist/bedAccessory.js
- /var/lib/homebridge/node_modules/homebridge-bed-control/dist/platform.js
- /var/lib/homebridge/node_modules/homebridge-bed-control/dist/index.js
- /var/lib/homebridge/node_modules/homebridge/lib/plugin.js
- /var/lib/homebridge/node_modules/homebridge/lib/pluginManager.js
- /var/lib/homebridge/node_modules/homebridge/lib/api.js
- /var/lib/homebridge/node_modules/homebridge/lib/childBridgeFork.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Function.Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/var/lib/homebridge/node_modules/homebridge-bed-control/src/bedAccessory.ts:1:1)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1024:12)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
[09/06/2024, 12:07:48] [homebridge-bed-control] Child bridge process ended
[09/06/2024, 12:07:48] [homebridge-bed-control] Process Ended. Code: 1, Signal: null
[09/06/2024, 12:07:50] [MySmartBlindsBridge] Refresh blind Stairwell Left Stairwell
[09/06/2024, 12:07:55] [homebridge-bed-control] Restarting Process...
[09/06/2024, 12:07:55] [MySmartBlindsBridge] STATUS: Stairwell Left Stairwell updateCurrentPosition : 66 (Actual 66)
[09/06/2024, 12:07:55] [MySmartBlindsBridge] Refresh blind Kitchen Kitchen Sink
[09/06/2024, 12:07:55] [MySmartBlindsBridge] Refresh blind Dining Room Left Dining Room
[09/06/2024, 12:07:56] [homebridge-bed-control] Launched child bridge with PID 6818
Error: Cannot find module 'hap-nodejs/dist/lib/definitions'
Require stack:
- /var/lib/homebridge/node_modules/homebridge-bed-control/dist/bedAccessory.js
- /var/lib/homebridge/node_modules/homebridge-bed-control/dist/platform.js
- /var/lib/homebridge/node_modules/homebridge-bed-control/dist/index.js
- /var/lib/homebridge/node_modules/homebridge/lib/plugin.js
- /var/lib/homebridge/node_modules/homebridge/lib/pluginManager.js
- /var/lib/homebridge/node_modules/homebridge/lib/api.js
- /var/lib/homebridge/node_modules/homebridge/lib/childBridgeFork.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1145:15)
    at Function.Module._load (node:internal/modules/cjs/loader:986:27)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
    at require (node:internal/modules/helpers:179:18)
    at Object.<anonymous> (/var/lib/homebridge/node_modules/homebridge-bed-control/src/bedAccessory.ts:1:1)
    at Module._compile (node:internal/modules/cjs/loader:1358:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1416:10)
    at Module.load (node:internal/modules/cjs/loader:1208:32)
    at Function.Module._load (node:internal/modules/cjs/loader:1024:12)
    at Module.require (node:internal/modules/cjs/loader:1233:19)
[09/06/2024, 12:07:56] [homebridge-bed-control] Child bridge process ended

```

**Plugin Config:**

```json
Show your Homebridge config.json here, remove any sensitive information.
```

**Screenshots:**

**Environment:**

* **Plugin Version**: v1.3.1
* **Homebridge Version**: v1.8.2 
* **Node.js Version**: v 20.14.0 
* **NPM Version**: 1.7.0 
* **Operating System**: Raspian 1.0.39

**Comment by dficken** [comment]
Seeing the same problem

**Comment by hbblebc** [comment]
checking on this

**Comment by randymassey** [comment]
👍👍

**Comment by hbblebc** [comment]
Okay try out 1.3.2 and let me know if it is working for you.

**Comment by randymassey** [comment]
So, I have done a quiz UD to 1.3.2 & the child bridge cannot load.Here is the red error segment in the log that relates to the plugin, without comparing, it does seem different from the original log data…Randy[11/06/2024, 23:01:52] ====================[11/06/2024, 23:01:52] ERROR LOADING PLUGIN homebridge-bed-control:[11/06/2024, 23:01:52] Error: Cannot find module '/var/lib/homebridge/node_modules/homebridge-bed-control/dist/settings' imported from /var/lib/homebridge/node_modules/homebridge-bed-control/dist/index.js    at finalizeResolution (node:internal/modules/esm/resolve:265:11)    at moduleResolve (node:internal/modules/esm/resolve:933:10)    at defaultResolve (node:internal/modules/esm/resolve:1157:11)    at ModuleLoader.defaultResolve (node:internal/modules/esm/loader:383:12)    at ModuleLoader.resolve (node:internal/modules/esm/loader:352:25)    at ModuleLoader.getModuleJob (node:internal/modules/esm/loader:227:38)    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:87:39)    at link (node:internal/modules/esm/module_job:86:36)[11/06/2024, 23:01:52] ====================Sent from my iPadOn Jun 11, 2024, at 9:49 PM, hbblebc ***@***.***> wrote:﻿
Okay try out 1.3.2 and let me know if it is working for you.

**Comment by randymassey** [comment]
Oh & back-stepping to 1.2.3 is still working & it boots normally & sees the configured items & responds to Siri request on HomePods etc..RandySent from my iPadOn Jun 11, 2024, at 9:49 PM, hbblebc ***@***.***> wrote:﻿
Okay try out 1.3.2 and let me know if it is working for you.

**Comment by hbblebc** [comment]
I am seeing the same error, although I am not sure what is causing it. I will update here when I have it fixed.

**Comment by randymassey** [comment]
👍👍Truly thanks for your efforts.Also quickly, once it is up and running again, I will also go in and test the subtraction of unused items, I never opened a bug report, but on my Sleep Number I seem to recall it was showing an outlet or something, and while I was able to Successfully subtract all of the things that I really needed to, there was one or two that just would not subtract in the config. It said something like it couldn’t find it or I can’t quite recall now as it’s been some time, but I see  from the new readme, you did some fixes in that area, I’ll take a look at that as well and report back if that issue was also resolved… I think because it was one item, I was able to sort of hide it and not have to contend with it.My bestRandySent from my iPhoneOn Jun 11, 2024, at 11:20 PM, hbblebc ***@***.***> wrote:﻿
I am seeing the same error, although I am not sure what is causing it. I will update here when I have it fixed.

**Comment by hbblebc** [comment]
Should be good to go now. Let me know once you test out the ignore list and file a ticket (or update #8) if you see anything weird.

**Comment by randymassey** [comment]
Def better, as far as booting & it’s child bridge coming up & Home app setting / reading values.I’m randomly getting slow down warnings in the log referring to the plugin & several items showed in the home app as there & active with a value in the room, but also under then under the 3 dots/circle top right in home app, in the non responsive list of devices at the top.  It’s shifting in & out & appearing in both places, but with values.  So they are not saying no response, which is odd.I do also see one of the items I could not disable before now has a non respond in it’s tile in Apple home & the other is the outlet my model doesn’t have & had it to, the one I just hid & left in, it now also has non response which is new.  Without being in the config, seems like it’s maybe more accurately identifying my i8 adjustable platform better possibly??  I know more when I get into the config & subtractions etc, which I did not do tonight, as it’s 11PM here in so cal, this was a quick boot up or not test to quickly report back.Tomorrow afternoon, I will have some time to get into the full web admin, go back to the subtraction list, did save in an RTF file the ID numbers for those specific pieces that were not successfully subtracting previously, and I believe it really was only like two of them, but I will absolutely test those and if everything‘s great, I’ll follow up with the reply here, but if there is a problem, I’ll do as you articulated in your message and open a ticket for that separately.… I will also see how many of the red slow down warnings appear in the log, I have, of course seen them in the past and the 1.2.3 version of the plug-in, but not as many as I saw after I installed it and rebooted to the latest tonight.so, confirmation of successful boot and ability to control with Siri for a fast test tonight, I will follow up on the details pertaining to the other topics tomorrow when I get some time in the afternoon.  Regards…RandySent from my iPhoneOn Jun 12, 2024, at 7:19 PM, hbblebc ***@***.***> wrote:﻿
Should be good to go now. Let me know once you test out the ignore list and file a ticket (or update #8) if you see anything weird.

**Comment by hbblebc** [comment]
I found another bug that might be related to the errors you are seeing, something is causing the outlet ID values to get mixed up. I am tracking it in #10.

**Comment by randymassey** [comment]
Understood.When I check in the afternoon well today now Lololol.I’ll look for any plugin updates first as well in case you revise before I start.RMSent from my iPhoneOn Jun 12, 2024, at 11:33 PM, hbblebc ***@***.***> wrote:﻿
I found another bug that might be related to the errors you are seeing, something is causing the outlet ID values to get mixed up. I am tracking it in #10.

**Comment by hbblebc** [comment]
Okay, as of 1.3.7 I am finally running error free on my end. Let me know what you see.

**Comment by randymassey** [comment]
So far so good on 1.3.8 after a couple of days.  Still running behind on the subtract topic but I will get to that.  The logs also look a lot cleaner in general as far as slow down complaints, in fact, I am not seeing any complaints so far anywhere in the logs about the plugin causing slowdowns like might originally appear after several days of use, & the issue with non response in the home app & both showing as active & non responsive is most def resolved.  

Since I installed the following morning, I went directly to the 1.3.8 version.

I have unexpected guests & a fully weekend, so it may not be until top of next week when I get time to go back in the config & re-visit the subtractions..  

Wonderful job, thank you for your efforts for this community...

Randy
